main.go

```diff
@@ -106,6 +112,14 @@ func main() {
 	srv := &http.Server{
 		Addr:    fmt.Sprintf(":%d", c.httpPort),
 		Handler: r,
+		// the maximum duration for reading the entire request, including the body
+		ReadTimeout: 2 * time.Second,
+		// the maximum duration before timing out writes of the response
```

Comment:
not entirely sure of the context here, but these timeouts seem way too short if they're for public-internet requests

Reply:
Yes, this might be short. I have increased. Found some article about this: https://blog.cloudflare.com/exposing-go-on-the-internet/

On outside, we just accept feedback in json format. So there should not be much of data to read. Also response is very simple. In that mind, we should not tolerate pushing a lot of data or give a lot of time to read big requests.

Reply:
It would be better to limit by size if that's the actual goal?

Some people have slow internet connections...so a malicious actor on a high-bandwidth connection will always be able to send way more data than a legitimate user on a slow connection (if trying to limit bandwidth via "time").

Can you estimate "not much data" in bytes?

Reply:
Currently data payload it is limited to 32kb. For feedback to receive it is more then enough.
Timeout is needed to mitigate this type of vulnerability: https://deepsource.io/directory/analyzers/go/issues/GO-S2112

Overall having timeout values makes sense, question, what values works best. As we accept small data (max 32kb), so even with slow network, this should be achievable in current times.

Reply:
Looks good to me. I don't know what timeout values are the right ones for slower internet connections..

Reply:
We can deploy on vps and test. If necessary, we can increase values

Reply:
How would you know if someone has tried to submit feedback, but been denied because of a too-aggressive timeout?

Ideally, this would be its own ticket + PR instead of "repository improvements" which doesn't discuss limits at all.

If you do want to limit by size, do that: it's way more accurate anyway.

I am aware of the "slowloris" attack but the real risk here seems to be throwing away "actual feedback". If someone wants to "attack" destiny or winden, they're not going to "slowloris" the feedback API, they're going to destroy all mailboxes every few seconds, making the service unusable.

Reply:
I don't think these values are too-aggressive, but you are welcome to suggest better values, however I still would like to have timeout setup even if it is little improvement for security. We will perform some testing and we can always increase them.

As for monitoring, yes, we don't have anything on application side to report if something doesn't work with client. Some discussion can be found here: https://github.com/LeastAuthority/it-ops/issues/64 
However, here we could log at least failed responses on server side, when 4xx or 5xx happening.

Mailbox issues and improvements should be discussed in a separate repo.

Reply:
I don't think we have a proper basis to decide either way. We should have a proper ticket to discuss them, though, instead of PR about unrelated things.